adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,41 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source){
+    std::stack<char> valid;
+
+    for(const char & c : source){
+        
+        for(const auto & b : brackets ){
+            if(c==b.second && valid.empty()){
+                return false;
+            }
+
+            if(c==b.first){
+                valid.push(c);
+                break;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es77-minimize-the-use-of-break-and-continue-in-loops

try to rewrite and simplify this function, hiding the inner loop in helper